app/code/Magento/MediaStorage/App/Media.php

```diff
@@ -190,7 +190,8 @@ public function launch(): ResponseInterface
             $fileAbsolutePath = $this->directoryPub->getAbsolutePath($this->relativeFileName);
             $fileRelativePath = str_replace(rtrim($this->mediaDirectoryPath, '/') . '/', '', $fileAbsolutePath);
             if (!$isAllowed($fileRelativePath, $allowedResources)) {
-                throw new LogicException('The path is not allowed: ' . $this->relativeFileName);
+                require_once 'errors/404.php';
+                exit;
```

Comment:
I believe we shouldn't remove the exception from here, instead, we should handle it properly where we believe that the 404 should be displayed. Other clients that use this function may react differently than show 404 and exit.

Reply:
The code is fully encapsulated in the [\Magento\Framework\App\Bootstrap](https://github.com/magento/magento2/blob/2.4-develop/lib/internal/Magento/Framework/App/Bootstrap.php#L259-L273) class, so there is no way to handle this exception from [pub/get.php](https://github.com/magento/magento2/blob/2.4-develop/pub/get.php#L87).